components/infobox/wikis/starcraft2/infobox_unit_custom.lua

```diff
@@ -0,0 +1,313 @@
+local _SUPPLY = mw.loadData('Module:Supply')
+local _HP = '[[File:Icon_Hitpoints.png|link=]]'
+local _SHIELDS = '[[File:Icon_Shields.png|link=Plasma Shield]]'
+local _ARMOR = '[[File:Icon_Armor.png|link=Armor]]'
+
+local _args
+local _race
+
+function CustomUnit.run(frame)
+	local unit = Unit(frame)
+	_args = unit.args
+	unit.nameDisplay = CustomUnit.nameDisplay
+	unit.setLpdbData = CustomUnit.setLpdbData
+	unit.createWidgetInjector = CustomUnit.createWidgetInjector
+	return unit:createInfobox(frame)
+end
+
+function CustomInjector:addCustomCells()
+	local widgets = {
+		Cell{name = 'Attributes', content = {_args.attributes}},
+		Cell{name = 'Energy', content = {_args.energy}},
+		Cell{name = 'Sight', content = {_args.sight}},
+		Cell{name = 'Detection range', content = {_args.detection_range}},
+		Cell{name = 'Speed', content = {_args.speed}},
+		Cell{name = 'Speed Multiplier on Creep', content = {_args.creepspeedmult}},
+		Cell{name = 'Speed on Creep', content = {_args.speedoncreep}},
+		Cell{name = 'Flags', content = {_args.flags}},
+		Cell{name = 'Morphs into', content = {_args.morphs}},
+		Cell{name = 'Cargo size', content = {_args.cargo_size}},
+		Cell{name = 'Cargo capacity', content = {_args.cargo_capacity}},
+		Cell{name = 'Strong against', content = {CustomUnit:_convertToList(_args.strong)}},
+		Cell{name = 'Weak against', content = {CustomUnit:_convertToList(_args.weak)}},
+	}
+
+	if _args.game ~= 'lotv' and _args.buildtime then
+		table.insert(widgets, Center{content = {
+			'<small>\'\'\'Note:\'\'\' ' ..
+			'All time-related values are expressed assuming Normal speed, as they were before LotV.' ..
+			' \'\'See [[Game Speed]].\'\'</small>'
+		}})
+	end
+
+	return widgets
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'cost' and not String.isEmpty(_args.min) then
+		return {
+			Cell{name = 'Cost', content = {CustomUnit:_getCostDisplay()}},
+		}
+	elseif id == 'requirements' then
+		return {
+			Cell{name = 'Requirements', content = {CustomUnit:_convertToList(_args.requires)}},
+		}
+	elseif id == 'hotkey' then
+		return {
+			Cell{name = '[[Hotkeys per Race|Hotkey]]', content = {CustomUnit:_getHotkeys()}}
+		}
+	elseif id == 'type' and (not String.isEmpty(_args.size) or not String.isEmpty(_args.type)) then
+		local display = _args.size
+		if not display then
+			display = _args.type
+		elseif _args.type then
+			display = display .. ' ' .. _args.type
+		end
+		return {
+			Cell{name = 'Type', content = {display}}
+		}
+	elseif id == 'defense' then
+		return {
+			Cell{name = 'Defense', content = {CustomUnit:_defenseDisplay()}}
+		}
+	elseif id == 'attack' then
+		local attacks = {}
+		local index = 1
+		while not String.isEmpty(_args['attack' .. index .. '_target']) do
+			table.insert(attacks, CustomUnit:_getAttack(index))
+		end
+		return attacks
+	end
+	return widgets
+end
+
+function CustomUnit:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomUnit:_convertToList(strg)
```

Comment:
Wouldn't be a bad shout for something like Module:String or Module:Array

Reply:
adjusted it a bit to enable different delimiters
```lua
function String.convertWikiListToHtmlList(str, delimiter)
	if String.isEmpty(str) then
		return ''
	end
	if String.isEmpty(delimiter) then
		delimiter = '*'
	end
	local strArray = mw.text.split(str, delimiter)
	local list = mw.html.create('ul')
	for _, item in ipairs(strArray) do
		if not String.isEmpty(item)
			list:tag('li'):wikitext(item)
		end
	end
	return tostring(list)
end
```
imo it is a string function (entry is a string (+ delimiter) and out, hence i will paste it (+adjust the /doc) in the string module(s) if you are okay with that
will upload several modules (incl., the string one(s)) to the git later today
since this function is already used in several sc2 custom infoboxes i would open a pr (after the function is on the git) to adjust the modules that have that function